src/main/java/nextstep/subway/domain/Sections.java

```diff
@@ -50,40 +100,47 @@ public void add(Section section) {
         sections.add(section);
     }
 
-    public List<Station> getStations() {
-        if (isEmptySections()) {
-            return Collections.emptyList();
+    private void validateSection(Section section) {
+        if (hasBothUpAndDownStations(section)) {
+            throw SectionsAddException.ALREADY_BOTH_STATION_REGISTER_EXCEPTION();
         }
 
-        return getOrderedStations();
-    }
-
-    public void deleteLastSection(Station station) {
-        if (isEmptySections()) {
-            throw new SectionsException(ErrorCode.NOT_FOUND_SECTION_EXCEPTION);
+        if (hasNotSameStationIn(section)) {
+            throw SectionsAddException.NOT_FOUND_BOTH_STATION_EXCEPTION();
         }
+    }
 
-        Section lastSection = getLastSection();
-        if (lastSection.hasNotDownStation(station)) {
-            throw new SectionsException(ErrorCode.NOT_SAME_DOWN_STATION_EXCEPTION);
-        }
+    private boolean isPossibleAddFrontOrBack(Section section) {
+        return isSectionsUpStation(section.getDownStation()) || isSectionsDownStation(section.getUpStation());
+    }
 
-        sections.remove(lastSection);
+    private Section findTargetSection(Section section) {
+        return sections.stream()
+                .filter(s -> s.hasSameUpStation(section.getUpStation()) || s.hasSameDownStation(section.getDownStation()))
+                .findFirst()
+                .orElseThrow(() -> SectionsDeleteException.NOT_FOUND_LAST_SECTION_EXCEPTION());
```

Comment:
아래와 같은 스타일은 어떻게 생각하세요? 😄 

```suggestion
                .orElseThrow(SectionsDeleteException::NOT_FOUND_LAST_SECTION_EXCEPTION);
```